### ISSUE TITLE
"Organize imports" removes used imports in some cases

### ISSUE THREAD

**Issue by nadako** [vshaxe/vshaxe#429]
The unused import feature is quite useful, however with large codebases it can sometimes work incorrectly and break code:

Because it depends on the diagnostics data it might happen that an user edited the file, making some imports used (and some other unused), pressed save and then organize imports right after. The new diagnostics data was requested (by saving), but have not returned yet, and the organize imports will use the obsolete data and remove wrong imports.

I would suggest "organize imports" to wait for diagnostics response if there is one in the queue. And also MAYBE disable the feature for dirty/unsaved files just to be safe.

**Comment by Gama11** [comment]
>And also MAYBE disable the feature for dirty/unsaved files just to be safe.

I think it should trigger a diagnostics request if necessary actually, and wait for the updated results.

**Comment by AlexHaxe** [comment]
somewhat connected: when you have broken code some imports might also be marked as unused and will be removed (https://github.com/HaxeFoundation/haxe/issues/8988)

**Comment by Gama11** [comment]
Yeah, unused import diagnostics are not reliable at all, so I only have `source.sortImports` active on save. :(

**Comment by nadako** [comment]
Well they are actually pretty reliable for us, it's just this one issue with outdated data.

**Comment by madneon** [comment]
Please also consider disabling auto-import in Settings: https://github.com/vshaxe/vshaxe/issues/630
